rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/util/RMProxyUserInterface.java

```diff
@@ -291,6 +295,40 @@ public AttributeList getMBeanAttributes(ObjectName name, String[] attributes)
         return this.jmxClient.getConnector().getMBeanServerConnection().getAttributes(name, attributes);
     }
 
+    /**
+     * Set a single JMX attribute of the MBean <code>objectName</code>.
+     * Only integer and string attributes are currently supported, see <code>type</code>.
+     *
+     * @param objectName    the object name of the MBean
+     * @param type          the type of the attribute to set ('int' and 'string' are currently supported)
+     * @param name          the name of the attribute to set
+     * @param value         the new value of the attribute (defined as a String, it is automatically converted according to <code>type</code>)
+     * @throws InstanceNotFoundException
+     * @throws IntrospectionException
+     * @throws ReflectionException
+     * @throws IOException
+     * @throws MBeanException
+     * @throws InvalidAttributeValueException
+     * @throws AttributeNotFoundException
+     */
+    public void setMBeanAttribute(ObjectName objectName, String type, String name, String value)
+            throws InstanceNotFoundException, IntrospectionException, ReflectionException, IOException, MBeanException,
+            InvalidAttributeValueException, AttributeNotFoundException {
+        Object attributeValue;
+        switch (type.toLowerCase()) {
+            case "int":
+                attributeValue = Integer.valueOf(value);
+                break;
+            case "string":
+                attributeValue = value;
+                break;
+            default:
+                return;
+        }
+        this.jmxClient.getConnector().getMBeanServerConnection().setAttribute(objectName,
```

Comment:
will potentials `AttributeNotFoundException` be translated into 404 status code from the REST interface ?

Reply:
Well I'm not even sure as most of the code was taken from the exiting `getNodeMBeansInfo` method and therefore I hope that all `InstanceNotFoundException`, `IntrospectionException`, `ReflectionException`, `IOException`, `NotConnectedException`, `MalformedObjectNameException`, and `NullPointerException` are mapped to HTTP response codes.
I'll test it anyway, do you know where REST errors are handled in general?

Reply:
I believe RESTEASY maps exceptions to status code 500. I don't know how to define custom error handlers with RESTEASY but in Spring you usually catch exceptions (`InstanceNotFoundException`, `IntrospectionException`, `ReflectionException`, `IOException`, `NotConnectedException`, `MalformedObjectNameException`) manually in the exposed service and throw your custom exception handler instead so you don't expose your internals. Don't bother changing it if that's the case, we don't support it in the other projects anyway. We can always improve that later, there's no hurry.